src/olympia/bandwagon/tests/test_views.py

```diff
@@ -249,6 +250,17 @@ def test_public(self):
         assert pq(r.content)('.meta .view-stats').length == 0, (
             'Only add-on authors can view stats')
 
+    def test_contributer(self):
+        with override_settings(COLLECTION_FEATURED_THEMES_ID=self.c.id):
+            self.c.listed = False
+            self.c.save()
+            self.assertLoginRedirects(self.client.get(self.url), self.url)
+            user = UserProfile.objects.get(email='fligtar@gmail.com')
+            self.grant_permission(user, 'Collections:Contribute')
+            self.client.login(email='fligtar@gmail.com')
```

Comment:
It'd be much more realistic if you could login the user already for the first `client.get()` call and only after you granted the permission try again, with the same user then.

Reply:
https://github.com/mozilla/addons-server/pull/8603/commits/0d352a701ae257128fd9e02e36611146fc35a8cb